src/pages/instances/forms/UploadExternalFormatFileForm.tsx

```diff
@@ -0,0 +1,334 @@
+      }
+
+      // If the image is already in raw format, remove the format conversion option
+      // this will optimise the conversion process since raw vm images do not need to be converted
+      const isRawImage = await isImageTypeRaw(file);
+      if (isRawImage) {
+        await formik.setFieldValue("formatConversion", false);
+      }
+    }
+  };
+
+  const handleCloseModal = useCallback(() => {
+    formik.resetForm();
+    close();
+    notify.clear();
+  }, [formik.resetForm, close, notify]);
+
+  const noFileSelectedMessage = !formik.values.imageFile
+    ? "A file must be selected first."
+    : "";
+
+  return (
+    <>
+      <ScrollableContainer
+        className={classnames({
+          "fixed-height": fixedFormHeight,
+          "u-hide": uploadState,
+        })}
+        dependencies={[]}
+        belowIds={["modal-footer"]}
+      >
+        <Form onSubmit={formik.handleSubmit}>
+          <Input
+            id="image-file"
+            name="imageFile"
+            type="file"
+            label="External format (.qcow2, .vmdk, etc...)"
+            labelClassName="u-hide"
+            onChange={(e) => void handleFileChange(e)}
+          />
+          <Input
+            {...formik.getFieldProps("name")}
+            id="name"
+            type="text"
+            label="New instance name"
+            placeholder="Enter name"
+            error={formik.touched.name ? formik.errors.name : null}
+            disabled={!!noFileSelectedMessage}
+            title={noFileSelectedMessage}
+          />
+          <StoragePoolSelector
+            project={project?.name || ""}
+            value={formik.values.pool}
+            setValue={(value) => void formik.setFieldValue("pool", value)}
+            selectProps={{
+              id: "pool",
+              label: "Root storage pool",
+              disabled: !project || !!noFileSelectedMessage,
+              title: noFileSelectedMessage,
+            }}
+          />
+          <Select
+            {...formik.getFieldProps("architecture")}
+            id="architecture"
+            label="Image architecture"
+            options={archOptions}
+            disabled={!!noFileSelectedMessage}
+            title={noFileSelectedMessage}
+          />
+          <label htmlFor="">Conversion options</label>
+          <Input
+            {...formik.getFieldProps("formatConversion")}
+            type="checkbox"
+            label={
+              <>
+                Convert to raw format{" "}
+                <Icon
+                  name="information"
+                  title={
+                    noFileSelectedMessage
+                      ? noFileSelectedMessage
+                      : "Can be skipped if the image is already in raw format to speed up the import."
+                  }
+                />
+              </>
+            }
+            disabled={!!noFileSelectedMessage}
+            checked={formik.values.formatConversion}
+          />
```

Comment:
There is some clipping when the checkboxes are active. The blue border is cut on the left-hand side.

![Screenshot from 2024-09-11 18-06-07](https://github.com/user-attachments/assets/f138288c-0292-4452-9230-ea338f94ffac)
![Screenshot from 2024-09-11 18-06-02](https://github.com/user-attachments/assets/2a1b3f98-4f4f-488e-acc7-d50fb6f57c54)

Reply:
had to give the checkboxes a small `margin-left`

Reply:
This is a visual design crime, please don't misalign them with the rest of the modal. I think the root cause is the scrollable container in `UploadExternalFormatFileForm`. It also creates a half modal scrolling, see screen below. The same is true for the backup file. Maybe we should raise it to the `UploadInstanceFileModal` and not have scrollable container in the form components.

![image](https://github.com/user-attachments/assets/d36e7dcd-6040-4a90-9e45-0c86426ecdfb)

Reply:
I see, the issue with that is the modal buttons are inside the form components since they need access to the form submission handlers. Raising the scrollable container to the modal level would defeat the point since the aim is to always show the modal buttons and make form content scrollable. I can create a component for the file type radio selection and move that into the form components to deal with the half scrolling form problem. I think some padding on the form component will probably fix the check boxes without misalignment.

Reply:
Thanks for explaining, I now see the problem with that. Agree, your proposed solution sounds good.

Reply:
I've made the above changes, let me know what you think? :slightly_smiling_face: